Analysis/interface/TauSelection.h

```diff
@@ -0,0 +1,43 @@
+#include "TauMLTools/Analysis/interface/TauTuple.h"
+#include "TauMLTools/Core/interface/exception.h"
+
+#include "TLorentzVector.h"
+
+namespace analysis {
+    bool PassGenLeptonCut(const tau_tuple::Tau& tau)
```

Comment:
Perhaps `boost::optional<GenLeptonMatch> GetGenLeptonMatch(const tau_tuple::Tau& tau)` would be better